http-client/src/main/java/io/micronaut/http/client/netty/DefaultHttpClient.java

```diff
@@ -545,6 +546,16 @@ public void close() {
 
             @Override
             public <I, O, E> io.micronaut.http.HttpResponse<O> exchange(io.micronaut.http.HttpRequest<I> request, Argument<O> bodyType, Argument<E> errorType) {
+                if (!configuration.isAllowBlockEventLoop() && Thread.currentThread() instanceof FastThreadLocalThread) {
+                    throw new HttpClientException(
+                        "You are trying to run a BlockingHttpClient operation on a netty event " +
+                            "loop thread. This is a common cause for bugs: Event loops should " +
+                            "never be blocked. You can either mark your controller as " +
+                            "@ExecuteOn(TaskExecutors.BLOCKING), or use the reactive HTTP client " +
+                            "to resolve this bug. There is also a configuration option to " +
+                            "disable this check if you are certain a blocking operation is fine " +
+                            "here.");
```

Comment:
Use a java multiline string since we are Java 17 now

Reply:
it's a single line though

Reply:
https://openjdk.org/jeps/368#New-escape-sequences